src/library_openal.js

```diff
@@ -314,6 +314,21 @@ var LibraryOpenAL = {
       }
       {{{ makeSetValue('data', '0', '0', 'i32') }}};
       break;
+    case 0x1007 /* ALC_FREQUENCY */:
+      if (!device) {
+        AL.alcErr = 0xA001 /* ALC_INVALID_DEVICE */;
+        return 0;
+      }
+      {{{ makeSetValue('data', '0', '(new (window.AudioContext || window.webkitAudioContext)).sampleRate', 'i32') }}};
```

Comment:
Allocating a new `AudioContext` is an extremely heavy operation, and there is a limit to how many can be allocated, so we should not generate temp allocations of these objects lightly. In OpenAL specification in 6.3.8 (https://www.openal.org/documentation/openal-1.1-specification.pdf) it is stated that "For some tokens, NULL is a legal deviceHandle." However it is not specified whether `ALC_FREQUENCY`, `ALC_MONO_SOURCES` and/or `ALC_STEREO_SOURCES` require a legal device handle or if null is ok. Your implementation behaves as if null is not ok, do you have a specific reason for that? I'm trying to find the docs to understand the exact semantics of calling `alcGetInteger(ALC_FREQUENCY)`, but can't find a conclusive reference. Namely, does `alcGetInteger(ALC_FREQUENCY)` return the sampling frequency of the currently active context, or the default sampling frequency for newly created contexts if one was not specified at context creation time?

I believe it would be the former, so instead of allocating a new `AudioContext` here, the code could refer to `AL.currentContext` here to obtain the information. Does that make sense?